

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>T-Coffee Web Server &mdash; T-Coffee 11 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script type="text/javascript" src="_static/jquery.js"></script>
        <script type="text/javascript" src="_static/underscore.js"></script>
        <script type="text/javascript" src="_static/doctools.js"></script>
        <script type="text/javascript" src="_static/language_data.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="T-Coffee doc test" href="tcoffee_doctest.html" />
    <link rel="prev" title="T-Coffee Technical Documentation" href="tcoffee_technical_documentation.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="index.html" class="icon icon-home"> T-Coffee
          

          
          </a>

          
            
            
              <div class="version">
                11
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="tcoffee_installation.html">T-Coffee Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="tcoffee_quickstart_regressive.html">Quick Start Regressive Algorithm</a></li>
<li class="toctree-l1"><a class="reference internal" href="tcoffee_quickstart.html">Quick Start</a></li>
<li class="toctree-l1"><a class="reference internal" href="tcoffee_main_documentation.html">T-Coffee Main Documentation</a></li>
<li class="toctree-l1"><a class="reference internal" href="tcoffee_technical_documentation.html">T-Coffee Technical Documentation</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">T-Coffee Web Server</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#general">General</a></li>
<li class="toctree-l2"><a class="reference internal" href="#t-coffee-simple-msa">T-Coffee Simple MSA</a></li>
<li class="toctree-l2"><a class="reference internal" href="#protein-sequences">Protein Sequences</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#expresso-using-3d-structures">Expresso (using 3D structures)</a></li>
<li class="toctree-l3"><a class="reference internal" href="#m-coffee-combining-multiple-methods">M-Coffee (combining multiple methods)</a></li>
<li class="toctree-l3"><a class="reference internal" href="#psi-tm-coffee-transmembrane-proteins">PSI/TM-Coffee (transmembrane proteins)</a></li>
<li class="toctree-l3"><a class="reference internal" href="#psi-coffee-homology-extension">PSI-Coffee (homology extension)</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#rna-sequences">RNA Sequences</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#r-coffee-using-2d-prediction">R-Coffee (using 2D prediction)</a></li>
<li class="toctree-l3"><a class="reference internal" href="#sara-coffee-using-3d-structures">SARA-Coffee (using 3D structures)</a></li>
<li class="toctree-l3"><a class="reference internal" href="#rm-coffee-combining-multiple-methods">RM-Coffee (combining multiple methods)</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#dna-sequences">DNA Sequences</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#id1">M-Coffee (combining multiple methods)</a></li>
<li class="toctree-l3"><a class="reference internal" href="#pro-coffee-homologous-promoter-regions">Pro-Coffee (homologous promoter regions)</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#evaluation-tools">Evaluation Tools</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#tcs-transitive-consistency-score">TCS (Transitive Consistency Score)</a></li>
<li class="toctree-l3"><a class="reference internal" href="#irmsd-apdb-msa-structural-evaluation-under-maintenance">iRMSD/APDB (MSA structural evaluation) (under maintenance…)</a></li>
<li class="toctree-l3"><a class="reference internal" href="#t-rmsd-structural-clustering">T-RMSD (structural clustering)</a></li>
<li class="toctree-l3"><a class="reference internal" href="#strike-msa-evaluation-with-single-structure">STRIKE (MSA evaluation with single structure)</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="tcoffee_doctest.html">T-Coffee doc test</a></li>
<li class="toctree-l1"><a class="reference internal" href="tcoffee_faq.html">FAQ</a></li>
<li class="toctree-l1"><a class="reference internal" href="tcoffee_citations_references.html">Citations &amp; References</a></li>
<li class="toctree-l1"><a class="reference internal" href="tcoffee_contacts_adresses.html">Contacts, Adresses &amp; Contributions</a></li>
<li class="toctree-l1"><a class="reference internal" href="tcoffee_license.html">License &amp; Terms of Use</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">T-Coffee</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
      <li>T-Coffee Web Server</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="_sources/tcoffee_webserver.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="t-coffee-web-server">
<h1>T-Coffee Web Server<a class="headerlink" href="#t-coffee-web-server" title="Permalink to this headline">¶</a></h1>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>This chapter is currently under maintenance. Its aim is to describe what the T-Coffee web server is doing programmatically when you are using it; it provides technical details about the restrictions and commands in use.</p>
</div>
<p>In this chapter we describe briefly the available T-Coffee web servers and their usage (command lines and examples files); if you need more advanced options, you should use the T-Coffee package via command lines. The web server is a reduced version of the T-Coffee package, containing all T-Coffee modes for aligning protein, RNA or DNA sequences; it also contains the evaluation and downstream analysis tools (TCS, iRMSD/APDB, STRIKE, T-RMSD). Currently, all reformatting utilities are not available on the web server, however, you can choose some reformatting options related to the ouput format. Here we present briefly the webserver and the T-Coffee commands it contains.</p>
<div class="section" id="general">
<h2>General<a class="headerlink" href="#general" title="Permalink to this headline">¶</a></h2>
<p>Most of the following command lines used by the web server contains lots of different options called with flags; here is a summary of the options common to all comand lines:</p>
<blockquote>
<div><ul class="simple">
<li><p><strong>-in</strong>: your input file</p></li>
<li><p><strong>-output</strong>: specifies the output files you require</p></li>
<li><p><strong>-maxnseq</strong>: maximum number of sequences you can run (variable depending on the mode)</p></li>
<li><p><strong>-maxlength</strong>: maximum length of your sequences (variable depending on the mode)</p></li>
<li><p><strong>-case=upper</strong>: all residues/nucleotids will be upper case</p></li>
<li><p><strong>-seqnos=off</strong>: the size of the sequences is not indicated on the MSA</p></li>
<li><p><strong>-outorder=input</strong>: orders the sequences in the final MSA as in the input dataset</p></li>
<li><p><strong>-run_name</strong>: name of the job on the cluster</p></li>
<li><p><strong>-multi_core=4</strong>: uses only 4 cores on your server when running the job</p></li>
<li><p><strong>-quiet=stdout</strong>: standard verbose output</p></li>
</ul>
</div></blockquote>
</div>
<div class="section" id="t-coffee-simple-msa">
<h2>T-Coffee Simple MSA<a class="headerlink" href="#t-coffee-simple-msa" title="Permalink to this headline">¶</a></h2>
<p>The central part of the web server is the T-Coffee aligner, you can use it to align any kind of sequences (Protein, RNA or DNA alike). The command it runs is the following:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$#: t_coffee -in=data_93c5fbb0.in -mode=regular -output=score_html clustalw_aln fasta_aln \
    score_ascii phylip -maxnseq=150 -maxlen=10000 -case=upper -seqnos=off -outorder=input \
    -run_name=result -multi_core=4 -quiet=stdout
</pre></div>
</div>
</div>
<div class="section" id="protein-sequences">
<h2>Protein Sequences<a class="headerlink" href="#protein-sequences" title="Permalink to this headline">¶</a></h2>
<div class="section" id="expresso-using-3d-structures">
<h3>Expresso (using 3D structures)<a class="headerlink" href="#expresso-using-3d-structures" title="Permalink to this headline">¶</a></h3>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$#: t_coffee -in=data_93c5fbb0.in -mode=expresso -blast=LOCAL -pdb_db=/db/pdb/derived_data_\
    format/blast/2016-01-01/pdb_seqres.fa -evaluate_mode=t_coffee_slow -output=score_html \
    clustalw_aln fasta_aln score_ascii phylip -maxnseq=150 -maxlen=2500 -case=upper -seqnos= \
    off -outorder=input -run_name=result -multi_core=4 -quiet=stdout
</pre></div>
</div>
</div>
<div class="section" id="m-coffee-combining-multiple-methods">
<h3>M-Coffee (combining multiple methods)<a class="headerlink" href="#m-coffee-combining-multiple-methods" title="Permalink to this headline">¶</a></h3>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$#: t_coffee -in=data_93c5fbb0.in  Mpcma_msa Mmafft_msa Mclustalw_msa Mdialigntx_msa Mpoa_msa \
    Mmuscle_msa Mprobcons_msa Mt_coffee_msa -output=score_html clustalw_aln fasta_aln \
    score_ascii phylip -tree -maxnseq=150 -maxlen=2500 -case=upper -seqnos=off -outorder=input \
    -run_name=result -multi_core=4 -quiet=stdout
</pre></div>
</div>
</div>
<div class="section" id="psi-tm-coffee-transmembrane-proteins">
<h3>PSI/TM-Coffee (transmembrane proteins)<a class="headerlink" href="#psi-tm-coffee-transmembrane-proteins" title="Permalink to this headline">¶</a></h3>
<p>Two options are available (in addition to the choice of the database): without transmembrane prediction or with prediction (displayed color coded on the html file).</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$#: tmcoffee.sh -in data_9df741d4.in -mode psicoffee -blast_server LOCAL --search-db &#39;UniRef50 \
    -- Very Fast/Rough&#39; --search-type &#39;&#39; -prot_min_sim 50 -prot_max_sim 90 -prot_min_cov 70 \
    --search-out &#39;clustalw_aln fasta_aln score_ascii phylip score_html&#39; -maxnseq 1000 -maxlen \
    =5000 -case upper -seqnos=off -outorder input -run_name result -multi_core 4 -quiet=stdout

$#: tmcoffee.sh -in data_9df741d4.in -mode psicoffee -blast_server LOCAL --search-db &#39;UniRef50 \
    -- Very Fast/Rough&#39; --search-type &#39;transmembrane&#39; -prot_min_sim 50 -prot_max_sim 90 \
    -prot_min_cov 70 --search-out &#39;clustalw_aln fasta_aln score_ascii phylip score_html&#39;
    -maxnseq 1000 -maxlen 5000 -case upper -seqnos off -outorder input -run_name result º
    -multi_core 4 -quiet=stdout
</pre></div>
</div>
</div>
<div class="section" id="psi-coffee-homology-extension">
<h3>PSI-Coffee (homology extension)<a class="headerlink" href="#psi-coffee-homology-extension" title="Permalink to this headline">¶</a></h3>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$#: t_coffee -in=data_93c5fbb0.in -mode=psicoffee -blast=LOCAL -protein_db=/db/ncbi/201511/ \
    blast/db/nr.fa -output=score_html clustalw_aln fasta_aln score_ascii phylip -maxnseq=150 \
    -maxlen=2500 -case=upper -seqnos=off -outorder=input -run_name=result -multi_core=4 \
    -quiet=stdout
</pre></div>
</div>
</div>
</div>
<div class="section" id="rna-sequences">
<h2>RNA Sequences<a class="headerlink" href="#rna-sequences" title="Permalink to this headline">¶</a></h2>
<div class="section" id="r-coffee-using-2d-prediction">
<h3>R-Coffee (using 2D prediction)<a class="headerlink" href="#r-coffee-using-2d-prediction" title="Permalink to this headline">¶</a></h3>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$#: t_coffee -in=data_29091222.in -method=mafft_msa muscle_msa probconsRNA_msa -output= \
    score_html clustalw_aln fasta_aln score_ascii phylip -maxnseq=150 -maxlen=2500 -case=upper \
    -seqnos=off -outorder=input -run_name=result -tree -special_mode=rcoffee -method_limits= \
    consan_pair 5 150 -multi_core=4 -quiet=stdout
</pre></div>
</div>
</div>
<div class="section" id="sara-coffee-using-3d-structures">
<h3>SARA-Coffee (using 3D structures)<a class="headerlink" href="#sara-coffee-using-3d-structures" title="Permalink to this headline">¶</a></h3>
<p>SARA-Coffee is a bit complicated to run, it uses several third party packages and is run through a script and environment variables we set up; here is what it looks like:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>##: export X3DNA=/data/www-cn/sara_coffee_package/X3DNA;
##: export PDB_DIR=/data/www-cn/sara_coffee_package/PDBdir/;
##: export NO_REMOTE_PDB_DIR=1;
##: unset MAFFT_BINARIES;
##: cd $CACHE_4_TCOFFEE
##: ln -s /data/www-cn/sara_coffee_package/pdb_entry_type.txt);
$#: t_coffee -in data_3e6e7aec.in -method sara_pair -template_file \
    /data/www-cn/sara_coffee_package/TEMPLATEFILE,RNA -extend_mode rna2 -relax_lib 0 -transform \
    dna2rna -run_name=result -output score_html clustalw_aln -case=upper -seqnos=off -outorder= \
    input -multi_core=4 -pdb_min_sim 0 -quiet stdout
</pre></div>
</div>
</div>
<div class="section" id="rm-coffee-combining-multiple-methods">
<h3>RM-Coffee (combining multiple methods)<a class="headerlink" href="#rm-coffee-combining-multiple-methods" title="Permalink to this headline">¶</a></h3>
<p>Not yet available…</p>
</div>
</div>
<div class="section" id="dna-sequences">
<h2>DNA Sequences<a class="headerlink" href="#dna-sequences" title="Permalink to this headline">¶</a></h2>
<div class="section" id="id1">
<h3>M-Coffee (combining multiple methods)<a class="headerlink" href="#id1" title="Permalink to this headline">¶</a></h3>
<p>For now, M-Coffee by default is the same for DNA, RNA and protein sequences alike. There is no specific M-Coffee for DNA sequences.</p>
</div>
<div class="section" id="pro-coffee-homologous-promoter-regions">
<h3>Pro-Coffee (homologous promoter regions)<a class="headerlink" href="#pro-coffee-homologous-promoter-regions" title="Permalink to this headline">¶</a></h3>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$#: t_coffee -in=data_476efe5f.in -mode=procoffee -output=score_html clustalw_aln fasta_aln \
    score_ascii phylip -maxnseq=150 -maxlen=10000 -case=upper -seqnos=off -outorder=input \
    -run_name=result -multi_core=4 -quiet=stdout
</pre></div>
</div>
</div>
</div>
<div class="section" id="evaluation-tools">
<h2>Evaluation Tools<a class="headerlink" href="#evaluation-tools" title="Permalink to this headline">¶</a></h2>
<div class="section" id="tcs-transitive-consistency-score">
<h3>TCS (Transitive Consistency Score)<a class="headerlink" href="#tcs-transitive-consistency-score" title="Permalink to this headline">¶</a></h3>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="c1">##: tcs.sh -infile data_a98d61a6.in -in Mproba_pair -score 1 -output clustalw_aln fasta_aln \</span>
    <span class="n">phylip</span> <span class="n">score_ascii</span> <span class="n">tcs_weighted</span> <span class="n">tcs_replicate</span> <span class="n">score_html</span> <span class="o">-</span><span class="n">maxnseq</span> <span class="mi">1000</span> <span class="o">-</span><span class="n">maxlen</span> <span class="mi">8000</span> \
    <span class="o">-</span><span class="n">seqnos</span><span class="o">=</span><span class="n">off</span> <span class="o">-</span><span class="n">run_name</span> <span class="n">result</span> <span class="o">-</span><span class="n">multi_core</span> <span class="mi">4</span> <span class="o">--</span><span class="nb">filter</span><span class="o">-</span><span class="nb">type</span> <span class="n">column</span> <span class="o">--</span><span class="nb">filter</span><span class="o">-</span><span class="nb">min</span> <span class="mi">4</span> <span class="o">--</span><span class="nb">filter</span><span class="o">-</span><span class="nb">max</span> \
    <span class="mi">9</span> <span class="o">--</span><span class="nb">filter</span><span class="o">-</span><span class="n">gap</span> <span class="n">yes</span> <span class="o">-</span><span class="n">quiet</span><span class="o">=</span><span class="n">stdout</span>
</pre></div>
</div>
</div>
<div class="section" id="irmsd-apdb-msa-structural-evaluation-under-maintenance">
<h3>iRMSD/APDB (MSA structural evaluation) (under maintenance…)<a class="headerlink" href="#irmsd-apdb-msa-structural-evaluation-under-maintenance" title="Permalink to this headline">¶</a></h3>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$#: t_coffee -other_pg apdb -aln data_c7151320.in -apdb_outfile default -outfile default \
    -io_format hsg3 -output score_html -maximum_distance 10 -md_threshold 2.0 -similarity_ \
    threshold 70 -template_file EXPRESSO -run_name result -quiet stdout
</pre></div>
</div>
</div>
<div class="section" id="t-rmsd-structural-clustering">
<h3>T-RMSD (structural clustering)<a class="headerlink" href="#t-rmsd-structural-clustering" title="Permalink to this headline">¶</a></h3>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$#: t_coffee -in=data_b89d3438.in -mode=expresso -cache=$PWD -blast=LOCAL -pdb_db=/db/pdb/ \
    derived_data_format/blast/2016-01-01/pdb_seqres.fa -evaluate_mode=t_coffee_slow -output= \
    aln score_html -maxnseq=150 -maxlen=2500 -case=upper -outorder=input -run_name=result \
    -multi_core=4 -quiet=stdout; t_coffee -other_pg trmsd result.aln -template_file \
    result_pdb1.template_list -output color_html 2&gt;&amp;1; [ -e result.struc_tree.consensus ]
</pre></div>
</div>
</div>
<div class="section" id="strike-msa-evaluation-with-single-structure">
<h3>STRIKE (MSA evaluation with single structure)<a class="headerlink" href="#strike-msa-evaluation-with-single-structure" title="Permalink to this headline">¶</a></h3>
<dl>
<dt>::</dt><dd><p>$#: wget      <a class="reference external" href="ftp://ftp.rcsb.org/pub/pdb/derived_data/pdb_seqres.txt">ftp://ftp.rcsb.org/pub/pdb/derived_data/pdb_seqres.txt</a>
$#: install   <a class="reference external" href="ftp://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/LATEST">ftp://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/LATEST</a></p>
<p>$#: t_coffee -other_pg strike &lt;sequence file&gt; -template_file PDB  -pdb_db &lt;pdb_seqres&gt; -blast_server LOCAL</p>
</dd>
</dl>
</div>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="tcoffee_doctest.html" class="btn btn-neutral float-right" title="T-Coffee doc test" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="tcoffee_technical_documentation.html" class="btn btn-neutral float-left" title="T-Coffee Technical Documentation" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2018, Cedric Notredame

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>