

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Quick Start &mdash; T-Coffee 11 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script type="text/javascript" src="_static/jquery.js"></script>
        <script type="text/javascript" src="_static/underscore.js"></script>
        <script type="text/javascript" src="_static/doctools.js"></script>
        <script type="text/javascript" src="_static/language_data.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="T-Coffee Main Documentation" href="tcoffee_main_documentation.html" />
    <link rel="prev" title="Quick Start Regressive Algorithm" href="tcoffee_quickstart_regressive.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="index.html" class="icon icon-home"> T-Coffee
          

          
          </a>

          
            
            
              <div class="version">
                11
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="tcoffee_installation.html">T-Coffee Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="tcoffee_quickstart_regressive.html">Quick Start Regressive Algorithm</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Quick Start</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#basic-command-lines-or-modes">Basic Command Lines (or modes)</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#protein-sequences">Protein sequences</a></li>
<li class="toctree-l3"><a class="reference internal" href="#dna-sequences">DNA sequences</a></li>
<li class="toctree-l3"><a class="reference internal" href="#rna-sequences">RNA sequences</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#brief-overview-of-t-coffee-tools">Brief Overview of T-Coffee Tools</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#alignment-methods">Alignment methods</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#t-coffee">T-Coffee</a></li>
<li class="toctree-l4"><a class="reference internal" href="#m-coffee">M-Coffee</a></li>
<li class="toctree-l4"><a class="reference internal" href="#expresso">Expresso</a></li>
<li class="toctree-l4"><a class="reference internal" href="#r-coffee">R-Coffee</a></li>
<li class="toctree-l4"><a class="reference internal" href="#pro-coffee">Pro-Coffee</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="#evaluation-tools">Evaluation tools</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#tcs-msa-evaluation-based-on-consistency">TCS (MSA evaluation based on consistency)</a></li>
<li class="toctree-l4"><a class="reference internal" href="#irmsd-apdb-msa-structural-evaluation">iRMSD/APDB (MSA structural evaluation)</a></li>
<li class="toctree-l4"><a class="reference internal" href="#strike-single-structure-msa-evaluation">STRIKE (single structure MSA evaluation)</a></li>
<li class="toctree-l4"><a class="reference internal" href="#t-rmsd-structural-clustering">T-RMSD (structural clustering)</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#tutorial-practical-examples">Tutorial (Practical Examples)</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#introduction">Introduction</a></li>
<li class="toctree-l3"><a class="reference internal" href="#materials">Materials</a></li>
<li class="toctree-l3"><a class="reference internal" href="#procedures">Procedures</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="tcoffee_main_documentation.html">T-Coffee Main Documentation</a></li>
<li class="toctree-l1"><a class="reference internal" href="tcoffee_technical_documentation.html">T-Coffee Technical Documentation</a></li>
<li class="toctree-l1"><a class="reference internal" href="tcoffee_webserver.html">T-Coffee Web Server</a></li>
<li class="toctree-l1"><a class="reference internal" href="tcoffee_doctest.html">T-Coffee doc test</a></li>
<li class="toctree-l1"><a class="reference internal" href="tcoffee_faq.html">FAQ</a></li>
<li class="toctree-l1"><a class="reference internal" href="tcoffee_citations_references.html">Citations &amp; References</a></li>
<li class="toctree-l1"><a class="reference internal" href="tcoffee_contacts_adresses.html">Contacts, Adresses &amp; Contributions</a></li>
<li class="toctree-l1"><a class="reference internal" href="tcoffee_license.html">License &amp; Terms of Use</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">T-Coffee</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
      <li>Quick Start</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="_sources/tcoffee_quickstart.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="quick-start">
<h1>Quick Start<a class="headerlink" href="#quick-start" title="Permalink to this headline">¶</a></h1>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>This chapter has been extensively updated in 11/2016. All T-Coffee modes/tools/commands should be valid for version 9.03 and above, but it is not guaranteed for lower versions of T-Coffee.</p>
</div>
<div class="section" id="basic-command-lines-or-modes">
<h2>Basic Command Lines (or modes)<a class="headerlink" href="#basic-command-lines-or-modes" title="Permalink to this headline">¶</a></h2>
<p>This chapter is a quick overview on how to run T-Coffee alignment using predefined procedures we call “modes”. All the files mentioned can be found <a class="reference external" href="https://github.com/cbcrg/tcoffee/tree/master/examples">here</a>. You can also use examples associated with their corresponding command lines from the subsection <strong>Tutorial (practical examples)</strong> published in Nature Protocols (2011). Refer to the section <strong>T-Coffee Manual</strong> for more technical details about T-Coffee usage and tools. Please, use the corresponding citation when using a specific mode or tools of T-Coffee (see <strong>References and Citations</strong>) otherwise cite T-Coffee.</p>
<div class="section" id="protein-sequences">
<h3>Protein sequences<a class="headerlink" href="#protein-sequences" title="Permalink to this headline">¶</a></h3>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>----------------------------------------------------------------------------
Default
----------------------------------------------------------------------------

$$: t_coffee sample_seq1.fasta

Citation: Notredame et al., JMB (2000)                      PMID:10964570
----------------------------------------------------------------------------
Fast
----------------------------------------------------------------------------

$$: t_coffee sample_seq1.fasta -mode quickaln

Citation: Notredame et al., JMB (2000)                      PMID:10964570
----------------------------------------------------------------------------
Consistent (M-Coffee combines the most common MSA packages)
----------------------------------------------------------------------------

$$: t_coffee sample_seq1.fasta -mode mcoffee

Citation: Wallace et al., Nucleic Acids Res. (2006)         PMID:16556910
----------------------------------------------------------------------------
Structure (Expresso finds structures homologous to your sequences)
----------------------------------------------------------------------------

$$: t_coffee sample_seq1.fasta -mode expresso

Citation: Armougom et al. Nucleic Acids Res. (2006)         PMID:16845081
----------------------------------------------------------------------------
Homology (PSI-Coffee enriches your dataset with homologous sequences)
----------------------------------------------------------------------------

$$: t_coffee sample_seq1.fasta -mode psicoffee

Citation: Chang et al., BMC Bioinformatics (2012)           PMID:22536955
----------------------------------------------------------------------------
Accurate (combines Structures and Homology)
----------------------------------------------------------------------------

$$: t_coffee sample_seq1.fasta -mode accurate

Citation: Notredame et al., JMB (2000)                      PMID:10964570
</pre></div>
</div>
</div>
<div class="section" id="dna-sequences">
<h3>DNA sequences<a class="headerlink" href="#dna-sequences" title="Permalink to this headline">¶</a></h3>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>----------------------------------------------------------------------------
Default
----------------------------------------------------------------------------

$$: t_coffee sample_dnaseq1.fasta

Citation: Notredame et al., JMB (2000)                      PMID:10964570
----------------------------------------------------------------------------
Functional (Pro-Coffee increases accuracy of functional DNA regions )
----------------------------------------------------------------------------

$$: t_coffee sample_dnaseq1.fasta -mode procoffee

Citation: Erb et al., Nucleic Acids Res. (2012)             PMID:22230796
</pre></div>
</div>
</div>
<div class="section" id="rna-sequences">
<h3>RNA sequences<a class="headerlink" href="#rna-sequences" title="Permalink to this headline">¶</a></h3>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>----------------------------------------------------------------------------
Default
----------------------------------------------------------------------------

$$: t_coffee sample_rnaseq1.fasta

Citation: Notredame et al., JMB (2000)                      PMID:10964570
----------------------------------------------------------------------------
Structure 2D (R-Coffee uses predicted secondary structures)
----------------------------------------------------------------------------

$$: t_coffee sample_rnaseq1.fasta -mode rcoffee

Citation: Wilm et al., Nucleic Acids Res. (2008)            PMID:18420654
----------------------------------------------------------------------------
Structure 3D (R-Coffee combined with Consan structural alignments)
----------------------------------------------------------------------------

$#: t_coffee sample_rnaseq1.fasta -mode rcoffee_consan

Citation: Wilm et al., Nucleic Acids Res. (2008)            PMID:18420654
----------------------------------------------------------------------------
Accurate (RM-Coffee use M-Coffee and secondary structure predictions)
----------------------------------------------------------------------------

$$: t_coffee sample_rnaseq1.fasta -mode rmcoffee

Citation: Wilm et al., Nucleic Acids Res. (2008)            PMID:18420654
</pre></div>
</div>
</div>
</div>
<div class="section" id="brief-overview-of-t-coffee-tools">
<h2>Brief Overview of T-Coffee Tools<a class="headerlink" href="#brief-overview-of-t-coffee-tools" title="Permalink to this headline">¶</a></h2>
<p>We only give you the very basics here, please go to the <strong>T-Coffee Main Documentation</strong> for a more detailed description. You can also try the <strong>T-Coffee tutorial</strong> for a practical training on T-Coffee alignment and other tools using applied examples on published research data.</p>
<div class="section" id="alignment-methods">
<h3>Alignment methods<a class="headerlink" href="#alignment-methods" title="Permalink to this headline">¶</a></h3>
<div class="section" id="t-coffee">
<h4>T-Coffee<a class="headerlink" href="#t-coffee" title="Permalink to this headline">¶</a></h4>
<p>Write or copy all your sequences (protein, DNA or RNA) in a given text file using one of the following format: Swiss-Prot, FASTA or PIR. Run T-Coffee with the following command line:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$$: t_coffee sample_seq1.fasta
</pre></div>
</div>
<p>When aligning, T-Coffee will always at least generate three files:</p>
<blockquote>
<div><ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">sample_seq1.aln</span></code>  : Multiple Sequence Alignment (ClustalW format by default)</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">sample_seq1.dnd</span></code>  : guide tree (Newick format)</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">sample_seq1.html</span></code> : colored MSA according to consistency (html format)</p></li>
</ul>
</div></blockquote>
<p>In principle, the type of the sequences is automatically detected and the default methods adapted accordingly. Sometimes, however, this may fail either because the sequences are too short or contain too many ambiguity codes. When this happens, you are advised to explicitly set the type of your sequences using the flag <strong>-type</strong>.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$$: t_coffee sample_dnaseq1.fasta -type=dna
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Please cite: Notredame, C., Higgins, D.G., Heringa, J. T-Coffee: a novel method for fast and accurate multiple sequence alignment. J. Mol. Biol., 302(1):205-217 (2000), PMID:10964570 and/or Magis, C., Taly, J.-F., Bussotti, G., Chang, J.M., Di Tommaso, P., Erb, I., Espinosa-Carrasco, J., Notredame, C. <strong>T-Coffee: tree-based consistency objective function for alignment evaluation</strong>. Methods Mol. Biol., 1079:117-129 (2014), PMID:24170398</p>
</div>
</div>
<div class="section" id="m-coffee">
<h4>M-Coffee<a class="headerlink" href="#m-coffee" title="Permalink to this headline">¶</a></h4>
<p>M-Coffee is a meta version of T-Coffee that combines the output of eight aligners (MUSCLE, ProbCons, POA, DIALIGN-T, MAFFT, ClustalW, PCMA and T-Coffee); when installing T-Coffee, all required packages are automatically installed on your computer. To use M-Coffee, write your sequences in a file (format: Swiss-Prot, FASTA or PIR) and run the following command 1. M-Coffee is a predefined combination of different types of aligners; there is a faster version called fm-Coffee (command 2) which combines the fastest aligners (Kalign, MUSCLE and MAFFT). Finally, the user can make its own combination of aligners included in T-Coffee by specifying the list of packages to be combined; here is an example of T-Coffee combining ClustalW, Kalign and ProbCons (command 3).</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>Command 1: running M-Coffee
$$: t_coffee sample_seq1.fasta -mode mcoffee

Command 2: running fm-Coffee
$$: t_coffee sample_seq1.fasta -mode fmcoffee

Command 3: user defined multiple methodes
$$: t_coffee sample_seq1.fasta -method clustalw_pair, kalign_pair, probcons_pair
</pre></div>
</div>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>If the program starts complaining one package or the other is missing, this means you will have to go the hard way and install all these packages yourself…</p>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Please cite: Wallace, I.M., O’Sullivan, O., Higgins, D.G., Notredame, C. <strong>M-Coffee: combining multiple sequence alignment methods with T-Coffee</strong>. Nucleic Acids Res., 34(6):1692-1699 (2006), PMID:16556910</p>
</div>
</div>
<div class="section" id="expresso">
<h4>Expresso<a class="headerlink" href="#expresso" title="Permalink to this headline">¶</a></h4>
<p>The default installation of T-Coffee provides you with the EBI wublast.pl client required to run Expresso ) command 1). Using this, Expresso will BLAST your sequences against the PDB database, identify the best targets (by default X-RAY structures, minimum 35% identical to your sequences) and use them to align your proteins using a structural aligner. If all the required structural packages for Expresso are not installed or if you want to select another structural aligner, you can select the structural package you want to use, for instance, if can use TM-align rather than SAP (command 2).</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>Command 1:
$$: t_coffee sample_seq1.fasta -mode expresso

Command 2:
$$: t_coffee sample_seq1.fasta -template_file PDB -method TMalign_pair
</pre></div>
</div>
<p>This correspondence between sequences and structures (templates) is declared in a FASTA-like file we call template file. Expresso automatically generates the template file (<code class="docutils literal notranslate"><span class="pre">&lt;your</span> <span class="pre">file</span> <span class="pre">name&gt;_pdb1.template_list</span></code>) that can be reused for applications, but you can also provide your own with the following format. This template file should have the following format:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">&gt;</span> <span class="o">&lt;</span><span class="n">seq_name</span><span class="o">&gt;</span> <span class="n">_P_</span> <span class="o">&lt;</span><span class="n">PDB</span> <span class="n">structure</span> <span class="n">file</span> <span class="ow">or</span> <span class="n">name</span><span class="o">&gt;</span>

<span class="o">*******</span> <span class="n">sample_3Dseq1</span><span class="o">.</span><span class="n">template</span> <span class="o">*******</span>
<span class="o">&gt;</span><span class="n">TNFR10</span><span class="o">-</span><span class="mi">2</span>  <span class="n">_P_</span> <span class="mi">1</span><span class="n">D4V2</span><span class="o">.</span><span class="n">pdb</span>
<span class="o">&gt;</span><span class="n">TNFR10</span><span class="o">-</span><span class="mi">3</span>  <span class="n">_P_</span> <span class="mi">1</span><span class="n">D4V3</span><span class="o">.</span><span class="n">pdb</span>
<span class="o">...</span>
<span class="o">**************************************</span>
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Please cite: Armougom, F., Moretti, S., Poirot, O., Audic, S., Dumas, P., Schaeli, B., Keduas, V., Notredame. C. <strong>Expresso: automatic incorporation of structural information in multiple sequence alignments using 3D-Coffee</strong>. Nucleic Acids Res., 34:W604-W608 (2006), PMID:16845081</p>
</div>
</div>
<div class="section" id="r-coffee">
<h4>R-Coffee<a class="headerlink" href="#r-coffee" title="Permalink to this headline">¶</a></h4>
<p>R-Coffee can be used to align RNA sequences, using their RNApfold predicted secondary structures (command 1). The best results are obtained by using the Consan pairwise method. If you have Consan installed (under maintenance…), you get access to one of the most accurate mode of R-Coffee (command 2). This will only work if your sequences are short enough (less than 200 nucleotides). A good alternative is the rmcoffee mode (command 3) that will run MUSCLE, ProbCons4RNA and MAFFT and then use the secondary structures predicted by RNApfold. Finally, you can also select yourself which methods should be combined by R-Coffee (command 4).</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>Command 1: R-Coffee
$$: t_coffee sample_rnaseq1.fasta -mode rcoffee

Command 2: R-Coffee + Consan
$#: t_coffee sample_rnaseq1.fasta -mode rcoffee_consan

Command 3: RM-Coffee
$$: t_coffee sample_rnaseq1.fasta -mode rmcoffee

Command 4: user defined R-Coffee
$$: t_coffee sample_rnaseq1.fasta -mode rcoffee -method lalign_id_pair,slow_pair
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Please cite: Wilm, A., Higgins, D.G., Notredame, C. <strong>R-Coffee: a method for multiple alignment of non-coding RNA</strong>. Nucleic Acids Res., 36(9):e52 (2008), PMID:18420654</p>
</div>
</div>
<div class="section" id="pro-coffee">
<h4>Pro-Coffee<a class="headerlink" href="#pro-coffee" title="Permalink to this headline">¶</a></h4>
<p>Pro-Coffee is a particular mode of T-Coffee designed to align specific functional DNA sequences, in particular regulatory regions. To run Pro-Coffee by default, just use command 1. In order to adjust the quality of the alignment, Pro-Coffee allows you to modify gap penalties (gap-opening and/or gap-extension) with specific flags (command 2).</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>Command 1: Pro-Coffee default
$$: t_coffee sample_dnaseq1.fasta -mode procoffee

Command 2: Pro-Coffee with modified parameters
$$: t_coffee sample_dnaseq1.fasta -method promo_pair@EP@GOP@-60@GEP@-1
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Please cite: Erb, I., González-Vallinas, J.R., Bussotti, G., Blanco, E., Eyras, E., Notredame, C. <strong>Use of ChIP-Seq data for the design of a multiple promoter-alignment method</strong>. Nucleic Acids Res., 40(7):e52 (2012), PMID:22230796.</p>
</div>
</div>
</div>
<div class="section" id="evaluation-tools">
<h3>Evaluation tools<a class="headerlink" href="#evaluation-tools" title="Permalink to this headline">¶</a></h3>
<div class="section" id="tcs-msa-evaluation-based-on-consistency">
<h4>TCS (MSA evaluation based on consistency)<a class="headerlink" href="#tcs-msa-evaluation-based-on-consistency" title="Permalink to this headline">¶</a></h4>
<p>Transitive Consistency Score (TCS) is an alignment evaluation score that makes it possible to identify the most correct positions in an MSA. It has been shown that these positions are the most likely to be structuraly correct and also the most informative when estimating phylogenetic trees. The TCS evaluation and filtering procedure is implemented in the T-Coffee package and can be used to evaluate and filter any third party MSA (including T-Coffee MSA of course!).</p>
<p>It’s usage is a bit tricky as it comes with a lot of different options, go to the <strong>T-Coffee Main Documentation</strong>, section <strong>Evaluating Your Alignment</strong> to have all the details about TCS.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Please cite: Chang, J.-M., Di Tommaso, P., Notredame, C. <strong>TCS: A new multiple sequence alignment reliability measure to estimate alignment accuracy and improve phylogenetic tree reconstruction</strong>. Mol. Biol. Evol., 31(6), 1625–1637 (2014), PMID:24694831 and/or Chang, J.-M., Di Tommaso, P., Lefort, V., Gascuel, O., Notredame, C. <strong>TCS: a web server for multiple sequence alignment evaluation and phylogenetic reconstruction</strong>. Nucleic Acids Res., 43(W1):W3-6 (2015), PMID:25855806</p>
</div>
</div>
<div class="section" id="irmsd-apdb-msa-structural-evaluation">
<h4>iRMSD/APDB (MSA structural evaluation)<a class="headerlink" href="#irmsd-apdb-msa-structural-evaluation" title="Permalink to this headline">¶</a></h4>
<p>iRMSD/APDB is not an alignment tool, it is an evalution tool of a given alignment using structural information. All you need is a file containing the alignment of sequences with a known structure (“template file”; see Expresso). If you don’t provide a template file, these sequences must be named according to their PDB ID, followed by the chain index (1aabA for example). In the first example (command 1) names are different therefore it won’t deliver any result. In that case, you should declare the correspondence between sequences and structures using your own template file (command 2). All the sequences do not need to have a known structure, but at least two is required otherwise it won’t deliver any result.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>Command 1:
$$: t_coffee -other_pg irmsd sample_3Dseq1.aln

Command 2:
$$: t_coffee -other_pg irmsd sample_3Dseq1.aln -template_file sample_3Dseq1.template
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Please cite: Armougom, F., Moretti, S., Keduas, V., Notredame, C. <strong>The iRMSD: a local measure of sequence alignment accuracy using structural information</strong>. Bioinformatics, 22(14):e35-e39 (2006), PMID:16873492</p>
</div>
</div>
<div class="section" id="strike-single-structure-msa-evaluation">
<h4>STRIKE (single structure MSA evaluation)<a class="headerlink" href="#strike-single-structure-msa-evaluation" title="Permalink to this headline">¶</a></h4>
<p>Under maintenance on the webserver or the T-Coffee package…</p>
</div>
<div class="section" id="t-rmsd-structural-clustering">
<h4>T-RMSD (structural clustering)<a class="headerlink" href="#t-rmsd-structural-clustering" title="Permalink to this headline">¶</a></h4>
<p>T-RMSD is a structure based clustering method using the iRMSD to drive the structural clustering of your aligned sequences with an available structure. The T-RMSD supports all the parameters supported by iRMSD or APDB. To run T-RMSD, type:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$$: t_coffee -other_pg trmsd sample_3Dseq1.aln -template_file sample_3Dseq1.template
</pre></div>
</div>
<dl class="simple">
<dt>The program then outputs a series of files:</dt><dd><ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">sample_3Dseq1.struc_tree.list</span></code> : list of the trees associated with every position.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">sample_3Dseq1.struc_tree.html</span></code> : colored columns supporting the tree.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">sample_3Dseq1.struc_tree.consensus_output</span></code> : schematic display of the results.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">sample_3Dseq1.struc_tree.consensus</span></code> : final consensus structural tree.</p></li>
</ul>
</dd>
</dl>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Please cite: Magis, C., Stricher, F., van der Sloot, A.M., Serrano, L., Notredame, C. <strong>T-RMSD: a fine-grained, structure based classification method and its application to the functional characterization of TNF receptors</strong>. J. Mol. Biol., 400(3):605-617 (2010), PMID:20471393 and/or Magis, C., van der Sloot, A.M., Serrano, L., Notredame, C. <strong>An improved understanding of TNFL/TNFR interactions using structure-based classifications</strong>. Trends Biochem. Sci., 37(9):353-363 (2012), PMID:22789664</p>
</div>
</div>
</div>
</div>
<div class="section" id="tutorial-practical-examples">
<h2>Tutorial (Practical Examples)<a class="headerlink" href="#tutorial-practical-examples" title="Permalink to this headline">¶</a></h2>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>This documentation is merely a cheat-sheet that recapitulates the material and the command lines associated with the manual. This tutorial itself is adpated from the <a class="reference external" href="http://www.nature.com/nprot/journal/v6/n11/full/nprot.2011.393.html">T-Coffee Nature Protocols Article</a> that can be followed step by step on the following <a class="reference external" href="http://www.tcoffee.org/Projects/tcoffee/workshops/tcoffeetutorials/index.html">website</a></p>
</div>
<div class="section" id="introduction">
<h3>Introduction<a class="headerlink" href="#introduction" title="Permalink to this headline">¶</a></h3>
<p>T-Coffee is a versatile Multiple Sequence Alignment method suitable for aligning most types of biological sequences. The series of protocols presented here show how the package can be used to multiply align proteins, DNA and RNA sequences. The package is an open source freeware available from <a class="reference external" href="http://www.tcoffee.org">our website</a>.</p>
<p>There are several parts: 1) the protein section presents controlled cases for PSI-Coffee the homology extended mode suitable for remote homologues, Expresso the structure based multiple aligner and M-Coffee, a meta version able to combine several third party aligners into one, 2) we then show how the T-RMSD option can be used to produce a functionally informative structure based clustering, 3) RNA alignment procedures are shown for R-Coffee a mode that produces secondary structure based MSAs, 4) DNA alignments are illustrated with Pro-Coffee, a multiple aligner specific of promoter regions, 5) finally, the last section presents some of the many reformatting utilities bundled with T-Coffee.</p>
</div>
<div class="section" id="materials">
<h3>Materials<a class="headerlink" href="#materials" title="Permalink to this headline">¶</a></h3>
<p>The list of files (input and output) required by this protocol is available from <a class="reference external" href="http://www.tcoffee.org/Packages/NatureProtocols/NatureProtocolDataset.tar.gz">here</a>. They can be automatically retrieved using the following command:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$$: t_coffee -other_pg nature_protocol.pl
</pre></div>
</div>
<p>This will create 4 repertories containing the input sequences necessary for the protocols we report in this section. For each part, all command lines have been collected into the file README.sh.</p>
</div>
<div class="section" id="procedures">
<h3>Procedures<a class="headerlink" href="#procedures" title="Permalink to this headline">¶</a></h3>
<ul class="simple">
<li><p><a class="reference external" href="http://www.tcoffee.org/Projects/tcoffee/workshops/tcoffeetutorials/index.html">Full Tutorial</a></p></li>
<li><p><a class="reference external" href="http://www.tcoffee.org/Projects/tcoffee/workshops/tcoffeetutorials/installation.html">Installation</a></p></li>
<li><p><a class="reference external" href="http://www.tcoffee.org/Projects/tcoffee/workshops/tcoffeetutorials/protein-alignment.html">Protein Multiple Sequence Alignments</a></p></li>
<li><p><a class="reference external" href="http://www.tcoffee.org/Projects/tcoffee/workshops/tcoffeetutorials/rna-alignment.html">RNA Multiple Sequence Alignments</a></p></li>
<li><p><a class="reference external" href="http://www.tcoffee.org/Projects/tcoffee/workshops/tcoffeetutorials/promoter-alignment.html">Promoter alignments</a></p></li>
<li><p><a class="reference external" href="http://www.tcoffee.org/Projects/tcoffee/workshops/tcoffeetutorials/reformating.html">Reformat alignments</a></p></li>
</ul>
</div>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="tcoffee_main_documentation.html" class="btn btn-neutral float-right" title="T-Coffee Main Documentation" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="tcoffee_quickstart_regressive.html" class="btn btn-neutral float-left" title="Quick Start Regressive Algorithm" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2018, Cedric Notredame

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>