

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Citations &amp; References &mdash; T-Coffee 11 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script type="text/javascript" src="_static/jquery.js"></script>
        <script type="text/javascript" src="_static/underscore.js"></script>
        <script type="text/javascript" src="_static/doctools.js"></script>
        <script type="text/javascript" src="_static/language_data.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Contacts, Adresses &amp; Contributions" href="tcoffee_contacts_adresses.html" />
    <link rel="prev" title="FAQ" href="tcoffee_faq.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="index.html" class="icon icon-home"> T-Coffee
          

          
          </a>

          
            
            
              <div class="version">
                11
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="tcoffee_installation.html">T-Coffee Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="tcoffee_quickstart_regressive.html">Quick Start Regressive Algorithm</a></li>
<li class="toctree-l1"><a class="reference internal" href="tcoffee_quickstart.html">Quick Start</a></li>
<li class="toctree-l1"><a class="reference internal" href="tcoffee_main_documentation.html">T-Coffee Main Documentation</a></li>
<li class="toctree-l1"><a class="reference internal" href="tcoffee_technical_documentation.html">T-Coffee Technical Documentation</a></li>
<li class="toctree-l1"><a class="reference internal" href="tcoffee_webserver.html">T-Coffee Web Server</a></li>
<li class="toctree-l1"><a class="reference internal" href="tcoffee_doctest.html">T-Coffee doc test</a></li>
<li class="toctree-l1"><a class="reference internal" href="tcoffee_faq.html">FAQ</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Citations &amp; References</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#comparative-genomics-biology">Comparative Genomics &amp; Biology</a></li>
<li class="toctree-l2"><a class="reference internal" href="#t-coffee-alignment-methods">T-Coffee Alignment Methods</a></li>
<li class="toctree-l2"><a class="reference internal" href="#t-coffee-web-servers">T-Coffee Web Servers</a></li>
<li class="toctree-l2"><a class="reference internal" href="#t-coffee-tools-for-msa-evaluation">T-Coffee Tools For MSA Evaluation</a></li>
<li class="toctree-l2"><a class="reference internal" href="#reviews-briefings-books">Reviews, Briefings &amp; Books</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="tcoffee_contacts_adresses.html">Contacts, Adresses &amp; Contributions</a></li>
<li class="toctree-l1"><a class="reference internal" href="tcoffee_license.html">License &amp; Terms of Use</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">T-Coffee</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
      <li>Citations &amp; References</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="_sources/tcoffee_citations_references.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="citations-references">
<h1>Citations &amp; References<a class="headerlink" href="#citations-references" title="Permalink to this headline">¶</a></h1>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>It is important that you cite T-Coffee when you use it. Citing us is (almost) like giving us money: it helps us convincing our institutions that what we do is useful and that they should keep paying our salaries and deliver donuts to our offices from time to time (Not that they ever did it, but it would be nice anyway). Cite the approriate web server if you use one, otherwise, cite the original paper from 2000 (By the way…NOOO, it was never named ‘T-Coffee 2000’).</p>
</div>
<div class="section" id="comparative-genomics-biology">
<h2>Comparative Genomics &amp; Biology<a class="headerlink" href="#comparative-genomics-biology" title="Permalink to this headline">¶</a></h2>
<p>Erb, I., Notredame, C. <strong>How should we measure proportionality on relative gene expression data?</strong> Theory Biosci., 135(1-2):21-36 (2016), PMID26762323</p>
<p>Catuara-Solarz, S., Espinosa-Carrasco, J., Erb, I., Langohr, K., Gonzalez, J.R., Notredame, C., Dierssen, M. <strong>Combined treatment with environmental enrichment and (-)-epigallocatchin-3-gallate ameliorates learning deficits and hippocampal alterations in a mouse model of Down syndrome</strong>. eNeuro., 3(5)(2015) pii:ENEURO.0103, PMID:27844057</p>
<p>Di Tommaso, P., Palumbo, E., Chatzou, M., Prieto, P.,Heuer, M.L., Notredame, C. <strong>The impact of Docker containers on ther performance of genomic pipelines</strong>. PeerJ., 3:e1273 (2015), PMID:26421241</p>
<p>Catuara-Solarz, S., Espinosa-Carrasco, J., Erb, I., Langohr, K., Notredame, C., Gonzalez, J.R., Dierssen, M. <strong>Principal component analysis of the effects of environmental enrichment and (-)-epigallocatchin-3-gallate on age-associated learning deficits in a mouse model of Down syndrome</strong>. Front Behav. Neurosci., 9:330 (2015), PMID:26696850</p>
<p>Earl, D., Nguyen, N., Hickey, G., Harris, R.S., Fitzgerald, S., Beal, K. Seledtsov, I., Raney, B.J., Clawson, H., Kim, J., Kemena, C., Chang, J.-M., Erb, I., Poliakov, A., Hou, M., Herrero, J., Kent, W.J., Solovyev, V., Darling, A.E., Ma, J., Notredame, C., Brudno, M., Dubchak, I., Haussler, D., Paten, B. <strong>Alignathon: a competitive assessment of whole-geome alignment methods</strong>. Genome Res., 24(12):2077-2089 (2014), PMID:25273068</p>
<p>The mouse ENCODE Consortium. <strong>A comparative encyclopedia of DNA elements in the mouse genome</strong>.  Nature, 515(7527):355-364 (2014), PMID:25409824</p>
<p>Bussotti, G., Notredame, C., Enright, A.J. <strong>Detecting and comparing non-coding RNAs in the high-throughput era</strong>. Int. J. Mol. Sci., 14(8):15423-15458 (2013), PMID:23887659</p>
<p>Chang, J.-M., Taly, J.-F., Erb, I., Sung, T.Y., Hsu, W.L. Tang, C.Y., Notredame, C., Su, E.C. <strong>Efficient and interpretable prediction of protein functional classes by correspondence analysis and compact set relations</strong>. PLoS One, 8(10):e75542 (2013), PMID:24146760</p>
<p>Magis, C., van der Sloot, A.M., Serrano, L., Notredame, C. <strong>An improved understanding of TNFL/TNFR interactions using structure-based classifications</strong>. Trends Biochem. Sci., 37(9):353-363 (2012), PMID:22789664</p>
<p>Derrien, T., Johnson, R., Bussotti, G., Tanzer, A., Djebali, S., Tilgner, H., Guernec, G., Martin, D., Merkel, A., Knowles, D.G., Lagarde, J., Veeravalli, L., Ruan, X., Ruan, Y., Lassmann, T., Carninci, P., Brown, J.B., Lipovich, L., Gonzalez, J.M., Davis, C.A., Shiekhattar, R., Gingeras, T.R., Hubbard, T.J., Notredame, C., Harrow, J., Guigó, R. <strong>The GENCODE v7 catalog of human long noncoding RNAs: analysis of their gene structure, evolution and expression</strong>. Genome Res., 22(9):1775-1789 (2012), PMID:22955988</p>
<p>Breen, M.S., Kemena, C., Vlasov, P.K., Notredame, C., Kondrashov, F.A. <strong>Epistasis as the primary factor in molecular evolution</strong>. Nature, 490(7421), 535-538 (2012), PMID:23064225</p>
<p>Bussotti, G., Raineri, E., Erb, I., Zytnicki, M., Wilm, A., Beaudoing, E., Bucher, P., Notredame, C. <strong>BlastR-fast and accurate database searches for non-coding RNAs</strong>. Nucleic Acids Res., 39(16):6886-6895 (2011), PMID:21624887</p>
<p>Lin, H.N., Notredame, C., Chang, J.-M., Sung, T.Y., Hsu, W.L. <strong>Improving the alignment quality of consistency based aligners with an evaluation function using synonymous protein words</strong>. PLoS One, 6(12):e27872, (2011), PMID:22163274</p>
<p>Ørom, U.A., Derrien, T., Beringer, M., Gumireddy, K., Gardini, A., Bussotti, G., Lai, F., Zytnicki, M., Notredame, C., Huang, Q., Guigo, R., Shiekhattar, R. <strong>Long noncoding RNAs with enhancer-like function in human cells</strong>. Cell, 143(1):46-58 (2010), PMID:20887892</p>
<p>Coll, O., Villalba, A., Bussotti, G., Notredame, C., Gebauer, F. <strong>A novel, noncanonical mechanism of cytoplasmic polyadenylation operates in Drosophila embryogenesis</strong>. Genes Dev., 24(2):129-134 (2010), PMID:20080951</p>
<p>Rausch, T., Emde, A.K., Weese, D., Döring, A., Notredame, C., Reinert, K. <strong>Segment-based multiple sequence alignment</strong>. Bioinformatics, 24(16):i187-i192 (2008), PMID:18689823</p>
<p>Dietmann, S:, Park, J., Notredame, C., Heger, A., Lappe, M., Holm, L. <strong>A fully automatic evolutionary classification of protein folds: Dali Domain Dictionary version 3</strong>. Nucleic Acids Res., 29(1):55-57 (2001), PMID:11125048</p>
</div>
<div class="section" id="t-coffee-alignment-methods">
<h2>T-Coffee Alignment Methods<a class="headerlink" href="#t-coffee-alignment-methods" title="Permalink to this headline">¶</a></h2>
<p>Kemena, C., Bussotti, G., Capriotti, E., Marti-Renom, M.A., Noredame, C. <strong>Using tertiary structure for computation of highly accurate multiple RNA alignments with the SARA-Coffee package</strong>. Bioinformatics, 29(9):1112-1119 (2013), PMID:234490094</p>
<p>Erb, I., González-Vallinas, J.R., Bussotti, G., Blanco, E., Eyras, E., Notredame, C. <strong>Use of ChIP-Seq data for the design of a multiple promoter-alignment method</strong>. Nucleic Acids Res., 40(7):e52 (2012), PMID:22230796</p>
<p>Chang, J.-M., Di Tommaso, P., Taly, J.-F., Notredame, C. <strong>Accurate multiple sequence alignment of transmembrane proteins with PSI-Coffee</strong>. BMC Bioinformatics, 13 Suppl 4:S1 (2012). PMID:22536955</p>
<p>Di Tommaso, P., Orobitg, M., Guirado, F., Cores, F., Espinosa, T., Notredame, C. <strong>Cloud-Coffee: implementation of a parallel consistency-based multiple alignment algorithm in the T-Coffee package and its benchmarking on the Amazon Elastic-Cloud</strong>. Bioinformatics, 26(15):1903-1904 (2010). PMID:20605929</p>
<p>Wilm, A., Higgins, D.G., Notredame, C. <strong>R-Coffee: a method for multiple alignment of non-coding RNA</strong>. Nucleic Acids Res., 36(9):e52 (2008), PMID:18420654</p>
<p>Armougom, F., Moretti, S., Poirot, O., Audic, S., Dumas, P., Schaeli, B., Keduas, V., Notredame. C. <strong>Expresso: automatic incorporation of structural information in multiple sequence alignments using 3D-Coffee</strong>. Nucleic Acids Res., 34:W604-W608 (2006), PMID:16845081</p>
<p>Wallace, I.M., O’Sullivan, O., Higgins, D.G., Notredame, C. <strong>M-Coffee: combining multiple sequence alignment methods with T-Coffee</strong>. Nucleic Acids Res., 34(6):1692-1699 (2006), PMID:16556910</p>
<p>O’Sullivan, O., Suhre, K., Abergel, C., Higgins, D.G., Notredame, C. <strong>3DCoffee: combining protein sequences and structures within multiple sequence alignments</strong>. J. Mol. Biol., 340(2):385-395 (2004). PMID: 15201059</p>
<p>O’Sullivan, O., Zehnder, M., Higgins, D.G., Bucher, P., Grosdidier, A., Notredame, C. <strong>APDB: a novel measure for benchmarking sequence alignment methods without reference alignments</strong>. Bioinformatics, 19 Suppl 1:i215-i221 (2003). PMID:12855461</p>
<p>Notredame, C. <strong>Mocca: semi-automatic method for domain hunting</strong>. Bioinformatics, 17(4):373-374 (2001). PMID: 11301309</p>
<p>Notredame, C., Higgins, D.G., Heringa, J. <strong>T-Coffee: A novel method for fast and accurate multiple sequence alignment</strong>. J. Mol. Biol., 302(1):205-217 (2000). PMID: 10964570</p>
<p>Notredame, C., Holm, L., Higgins, D.G. <strong>COFFEE: an objective function for multiple sequence alignments</strong>. Bioinformatics, 14(5):407-422 (1998). PMID: 968205</p>
</div>
<div class="section" id="t-coffee-web-servers">
<h2>T-Coffee Web Servers<a class="headerlink" href="#t-coffee-web-servers" title="Permalink to this headline">¶</a></h2>
<p>Floden, E.W., Tommaso, P.D., Chatzou, M., Magis, C., Notredame, C., Chang, J.-M. <strong>PSI/TM-Coffee: a web server for fasta accurate multiple sequence alignments of regular and transmembrane proteins using homology extension on reduced databases</strong>. Nucleic Acids Res., 44(W1):W339-W343 (2016). PMID:27106060</p>
<p>Chang, J.-M.*, Tommaso P.*, Lefort, V., Gascuel, O., Notredame, C. <strong>TCS: a web server for multiple sequence alignment evaluation and phylogenetic reconstruction</strong>. Nucleic Acids Research. 43(W1):W3-6 (2015). PMID: (<a href="#id1"><span class="problematic" id="id2">*</span></a>These two authors contribute equally)</p>
<p>Di Tommaso, P., Bussotti, G., Kemena, C., Capriotti, E., Chatzou, M., Prieto, P., Notredame, C. <strong>SARA-Coffee web server, a tool for computation of RNA sequence and structure multiple alignments</strong>. Nucleic Acids Res., 42(Web Server issue):W356-W360 (2014). PMID:24972831</p>
<p>Magis, C., Di Tommaso, P., Notredame, C. <strong>T-RMSD: a web server for automated fine-grained protein structural classifcation</strong>. Nucleic Acids Res., 41(Web Server issue):W358-W362 (2013). PMID:23716642</p>
<p>Rius, J., Cores, F., Solsona, F., van Hemert, J.I., Koetsier, J., Notredame, C. <strong>A user-friendly web portal for T-Coffee on supercomputers</strong>. BMC Bioinformatics, 12:150 (2011). PMID:21569428</p>
<p>Di Tommaso, P., Moretti, S., Xenarios, I., Orobitg, M., Montanyola, A., Chang, J.-M., Taly, J.-F., Notredame, C. <strong>T-Coffee: a web server for the multiple sequence alignment of protein and RNA sequences using structural information and homology extension</strong>. Nucleic Acids Res., 39(Web Server issue):W13-W17 (2011). PMID:21558174</p>
<p>Moretti, S., Wilm, A., Higgins, D.G., Xenarios, I., Notredame, C. <strong>R-Coffee: a web server for accurately aligning noncoding RNA sequences</strong>., Nucleic Acids Res., 36(Web Server issue):W10-W13 (2008). PMID:18483080</p>
<p>Moretti, S., Armougom, F., Wallace, I.M., Higgins, D.G., Jongeneel, C.V., Notredame, C. <strong>The M-Coffee web server: a meta-method for computing multiple sequence alignments by combining alternative alignment methods</strong>. Nucleic Acids Res., 35(Web Server issue):W645-W648 (2007). PMID:17526519</p>
<p>Moretti, S., Reinier, F., Poirot, O., Armougom, F., Audic, S., Keduas, V., Notredame, C. <strong>PROTOGENE: turning amino acid alignments into bona fide CDS nucleotide alignments</strong>. Nucleic Acids Res., 34(Web server issue):W600-W603 (2006), PMID:16845080</p>
<p>Armougom, F., Moretti, S., Keduas, V., Notredame, C. <strong>The iRMSD: a local measure of sequence alignment accuracy using structural information</strong>. Bioinformatics, 22(14):e35-e39 (2006). PMID:16873492</p>
<p>Armougom, F., Poirot, O., Moretti, S., Higgins, D.G., Bucher, P., Keduas, V., Notredame, C. <strong>APDB: a web server to evaluate the accuracy of the sequence alignments using structural information</strong>. Bioinformatics, 22(19):2439-2440 (2006). PMID:17032685</p>
<p>Claude, J.B., Suhre, K., Notredame, C., Claverie, J.M., Abergel C. <strong>CaspR: a web server for automated molecular replacement using homology modelling</strong>. Nucleic Acids Res., 32(Web Server issue):W606-9 (2004). PMID:15215460</p>
<p>Poirot, O., Suhre, K., Abergel, C., O’Toole, E., Notredame, C. <strong>3DCoffee&#64;igs: a web server for combining sequences and structures into a multiple sequence alignment</strong>. Nucleic Acids Res., 32(Web Server issue):W37-40 (2004). PMID: 15215345</p>
<p>Poirot, O., O’Toole, E., Notredame, C. <strong>Tcoffee&#64;igs: a web server for computing, evaluating and combining multiple sequence alignments</strong>. Nucleic Acids Res., Jul 1;31(13):3503-6 (2003). PMID: 12824354</p>
</div>
<div class="section" id="t-coffee-tools-for-msa-evaluation">
<h2>T-Coffee Tools For MSA Evaluation<a class="headerlink" href="#t-coffee-tools-for-msa-evaluation" title="Permalink to this headline">¶</a></h2>
<p>Chang, J.-M., Tommaso, P., Notredame, C. <strong>TCS: a new Multiple Sequence Alignment reliability measure to estimate alignment accuracy and improve phylogenetic tree reconstruction</strong>. Molecular Biology and Evolution, 31(6):1625-37 (2014). PMID:24694831</p>
<p>Kemena, C., Taly, J.-F., Kleinjung, J., Notredame, C. <strong>STRIKE: evaluation of protein MSAs using a single 3D structure</strong>. Bioinformatics, 27(24):3385-3391 (2011). PMID:220339207</p>
<p>Magis, C., Stricher, F., van der Sloot, A.M., Serrano, L., Notredame, C. <strong>T-RMSD: a fine-grained, structure based classification method and its application to the functional characterization of TNF receptors</strong>. J. Mol. Biol., 400(3):605-617 (2010), PMID:20471393</p>
<p>Armougom, F., Moretti, S., Keduas, V., Notredame, C. <strong>The iRMSD: a local measure of sequence alignment accuracy using structural information</strong>. Bioinformatics, 22(14):e35-e39 (2006), PMID:16873492</p>
</div>
<div class="section" id="reviews-briefings-books">
<h2>Reviews, Briefings &amp; Books<a class="headerlink" href="#reviews-briefings-books" title="Permalink to this headline">¶</a></h2>
<p>Chatzou, M., Magis, C., Chang, J.-M., Kemena, C., Bussotti, G., Erb, I., Notredame, C. <strong>Multiple sequence alignment modeling: methods and applications</strong>. Brief Bioinform., pii: bbv099 (2015). PMID:26615024</p>
<p>Magis, C., Taly, J.-F., Bussotti, G., Chang, J.M., Di Tommaso, P., Erb, I., Espinosa-Carrasco, J., Notredame, C. <strong>T-Coffee: tree-based consistency objective function for alignment evaluation</strong>. Methods Mol. Biol., 1079:117-129 (2014). PMID:24170398</p>
<p>Taly, J.-F.*, Magis, C.*, Bussotti, G., Chang, J.-M., Di Tommaso, P., Erb, I., Espinosa-Carrasco, J., Kemena, C., Notredame, C. <strong>Using the T-Coffee package to build multiple sequence alignments of protein, RNA, DNA sequences and 3D structures</strong>. Nature Protocols, 1669-1682 (2011), PMID:21979275 (<a href="#id3"><span class="problematic" id="id4">*</span></a>These two authors contribute equally)</p>
<p>Notredame, C. <strong>Computing multiple sequence/structure alignments with the T-Coffee package</strong>. Curr. Protoc. Bioinformatics, Chapter 3, Unit 3.8.1-25 (2010). PMID:20205190</p>
<p>Kemena, C., Notredame, C. <strong>Upcoming challenges for multiple sequence alignment methods in the high-throughput era</strong>. Bioinformatics, 25(19):2455-2465 (2009). PMID:19648142</p>
<p>Notredame, C. <strong>Recent evolutions of multiple sequence alignment algorithms</strong>. PLoS Comput. Biol., 3(8):e123 (2007). PMID:17784778</p>
<p>Notredame, C., Suhre, K. <strong>Computing multiple sequence/structure alignments with the T-Coffee package</strong>. Curr. Protoc. Bioinformatics, Chapter 3:Unit 3.8 (2004), PMID:18428722</p>
<p>Notredame, C., Abergel, C. <strong>Using multiple alignment methods to assess the quality of genomic data analysis</strong>. In: Andrade MA, editor. Bioinformatics and genomes: current perspectives. Wymondham (UK): Horizon Scientific Press,30–50 (2003).</p>
<p>Notredame, C. <strong>Recent progress in multiple sequence alignment: a survey</strong>. Pharmacogenomics, 3(1):131-144 (2002). PMID:11966409</p>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="tcoffee_contacts_adresses.html" class="btn btn-neutral float-right" title="Contacts, Adresses &amp; Contributions" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="tcoffee_faq.html" class="btn btn-neutral float-left" title="FAQ" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2018, Cedric Notredame

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>